dashboard/app/models/backpack.rb

```diff
@@ -16,19 +16,20 @@
 #
 class Backpack < ApplicationRecord
   belongs_to :user, optional: true
+  belongs_to :game, optional: true
 
   # The projects table used to be named storage_apps. This column has not been renamed
   # to reflect the new table name, so an alias is used to clarify which table this ID maps to.
   alias_attribute :project_id, :storage_app_id
 
-  def self.find_or_create(user_id, ip)
-    backpack = find_by_user_id(user_id)
+  def self.find_or_create(user_id, game_id, ip)
+    backpack = find_by(user_id: user_id, game_id: game_id)
     unless backpack
-      # Create a project for this user's backpack
+      # Create a project for this user's backpack in the app_type ('Javalab' or 'Pythonlab') determined by game_id
```

Comment:
nit: I would remove `'Javalab' or 'Pythonlab'` so we don't have to update this list in the future. We also may enable it for web lab 2 soon.